### DISCUSSION TITLE
3.23.0

### DISCUSSION THREAD

**Discussion by zenangst** [zenangst/KeyboardCowboy#505] [Announcements]
![Release 3 23 0](https://github.com/zenangst/KeyboardCowboy/assets/57446/ee04cf2a-40aa-48cd-9195-b0d940c12363)

# **Snippet Magic**: Write more with less. 
### Tiny trigger words now unleash mighty workflow wonders – all at your command.

- **UI Sleuthing Upgrade**: Capture UI elements in Electron apps like a boss – Spotify, Slack, and Discord, now fully at your mercy.
- **Window Wrangling Wizardry**: Anchoring perfected. No more window pile-ups, just smooth resizing for a seamless view.
- **Clipboard Charms**: Text commands now wield the $PASTEBOARD sorcery, turning your clipboard contents into instant command fuel.
- **Sidebar Shuffle**: User Modes strut down to the sidebar's end, making room for spotlight-stealers.
- **Notification Glow-Up**: Alert aesthetics reimagined – stay informed with style.
- **Icon Evolution**: User Modes now flaunt an iconic new symbol of identity.
- **Icon Expansion Pack**: Sprinkling in a dash of pictographic pizzazz across the UI.
- **Pasteboard Prowess**: Across-the-board reliability boost for pasteboard-dependent commands — like magic, but realer.
- **Shortcut Sonic Boom**: Dive straight into recording mode for shortcuts — no extra clicks, just slick picks.
- **Duplicate Alert Declutter**: Internal tweaks squash notification echoes for a smoother info flow.
- **Swift Bundle Alerts**: Notifications vanish gracefully with a new shortcut summon – no lingering farewells.
- **Multi-Mode Mending**: Squashed the gremlins that bungled simultaneous mode mojo — harmony restored.
- **File Menu Flourish**: 'New Command' now strutting its stuff in the File menu – because every newbie deserves a grand entrance.
- **Bug Bash Extravaganza**: Swatted those pesky glitches for a buttery-smooth operation.

<hr /><em>This discussion was created from the release <a href='https://github.com/zenangst/KeyboardCowboy/releases/tag/3.23.0'>3.23.0</a>.</em>

**Comment by mgcobrien** [comment]
Is there any documentation on how Snippets work?

I don't really understand what they're used for and the Wiki doesn't list anything about them: https://github.com/zenangst/KeyboardCowboy/wiki/Triggers

**Reply by zenangst** [reply]
@mgcobrien Here's an example of a snippet I set up to work in the terminal.

<img width="1070" alt="image" src="https://github.com/user-attachments/assets/b2096fb1-cc01-4a2f-8e03-9b1e7e39e5b0">

When I type `ghprls`, the trigger activates and runs my commands from the workflow, just like you'd expect.

It removes the input from the trigger before proceeding.

In this case, it invokes the `gh` command with some parameters.

Since it's in the terminal, I end the workflow with the enter key, and voila... a terminal snippet.

Hope that helps!

(I'm gonna update the wiki to include snippets)
